SDK/MobileCenter/Microsoft.Azure.Mobile.Windows.Shared/Storage/Storage.cs

```diff
@@ -261,31 +233,40 @@ internal Storage(IStorageAdapter storageAdapter)
         /// <exception cref="StorageException"/>
         public async Task<string> GetLogsAsync(string channelName, int limit, List<Log> logs)
         {
-            await OpenDbAsync().ConfigureAwait(false);
+            await StartTaskAsync().ConfigureAwait(false);
             logs?.Clear();
             var retrievedLogs = new List<Log>();
             MobileCenterLog.Debug(MobileCenterLog.LogTag, $"Trying to get up to {limit} logs from storage for {channelName}");
             try
             {
-                // Create the query
-                var command = _storageAdapter.CreateCommand();
-                var channelParameter = command.CreateParameter();
-                channelParameter.ParameterName = "channelName";
-                channelParameter.Value = channelName;
-                var limitParameter = command.CreateParameter();
-                limitParameter.ParameterName = "limit";
-                limitParameter.Value = limit;
-                command.Parameters.Add(channelParameter);
-                command.Parameters.Add(limitParameter);
-                command.CommandText =
-                    $"SELECT {RowIdColumn},* FROM {Table} " +
-                    $"WHERE {ChannelColumn}=@{channelParameter.ParameterName} " +
-                    $"LIMIT @{limitParameter.ParameterName}";
-                command.Prepare();
-
-                // Execute the query
                 var idPairs = new List<Tuple<Guid?, long>>();
-                await RetrieveLogsAsync(command, channelName, retrievedLogs, idPairs).ConfigureAwait(false);
+                var failedToDeserializeALog = false;
+                var retrievedEntries =
+                    await _storageAdapter.GetAsync<LogEntry>(entry => entry.Channel == channelName, limit)
+                        .ConfigureAwait(false);
+                foreach (var entry in retrievedEntries)
+                {
+                    if (_pendingDbIdentifiers.Contains(entry.Id))
+                    {
+                        continue;
+                    }
+                    try
+                    {
+                        var log = LogSerializer.DeserializeLog(entry.Log);
+                        retrievedLogs.Add(log);
+                        idPairs.Add(Tuple.Create(log.Sid, Convert.ToInt64(entry.Id)));
+                    }
+                    catch (JsonException e)
+                    {
+                        MobileCenterLog.Error(MobileCenterLog.LogTag, "Cannot deserialize a log in storage", e);
+                        failedToDeserializeALog = true;
+                        await DeleteLogAsync(channelName, entry.Id).ConfigureAwait(false);
```

Comment:
>await DeleteLogAsync(channelName, entry.Id).ConfigureAwait(false); [](start = 24, length = 66)

Question: is it better to throw or ignore storage exception from DeleteLogAsync? What's your reasoning of throwing storage exception here?